### ISSUE TITLE
Support AES-256 encryption

### ISSUE THREAD

**Issue by sannihith1995** [hhrutter/pdfcpu#83]
Hi @hhrutter,

I've been using this for a while and it's been great.
Recently when I tried to decrypt a file which has attachments using UserPW, it shows this error
`getV: "V" must be one of 1,2,4`

and I was able to decrypt the same file using **_'qpdf'_** tools.
One thing was, the password has a **_'$'_** in it and I was able to decrypt using quotes and the same did not work with pdfcpu.

After decrypting the file, I was able to extract attachments from it using pdfcpu.

May I know what might be the cause?

**Comment by hhrutter** [comment]
Thanks for using pdfcpu 💚 
Ok, so there are really only 3 values for the encryption algorithm that we can accept in order to continue as far as the spec goes:

<img width="791" alt="Screenshot 2019-05-21 at 09 15 00" src="https://user-images.githubusercontent.com/11322155/58187728-a45e1780-7cb7-11e9-93a2-1f84d9b40587.png">



Why don't you go ahead and output **v** in this function of *crypto.go* like so and then rebuild and run `pdfcpu decrypt` over you file - then we'll know more, I'm, curious ;)


```
func getV(d Dict) (*int, error) {

	v := d.IntEntry("V")
               
        if v == nil {
                 return nil, errors.Errorf("getV: \"V\" is nil")
        }

	if *v != 1 && *v != 2 && *v != 4 {
		return nil, errors.Errorf("getV: \"V\" must be one of 1,2,4 - got:%d", *v)
	}

	return v, nil
}
```

**Comment by sannihith1995** [comment]
Hey @hhrutter,

Thanks for the response.
I tried with different combinations as you suggested and here are the outputs for it.

When I tried to print and see what's the value for '**v**', It printed
`824634478472`

When I tried to bypass getV function and return '**v**' as it is, I got this response
`length: "Length" 256 not supported`

Then I tried to set value for '**v**' as '**1**' and '**2**' and got the same response again.

And finally when i tried to set it to '**4**', I got this
`checkStmv: unsupported "StdCF" entry in "CF": supportedCFEntry: invalid entry "CFM"`

**Comment by hhrutter** [comment]
Are you sure you are not printing a pointer value (which is an address)?
You have to dereference the pointer in the Errorf statement (*v)

**Comment by sannihith1995** [comment]
I am sorry. I just missed to deference it.
I just tried to print the value of '**v**' and it is '**5**'

**Comment by hhrutter** [comment]
This is interesting because 5 is not a defined value for v.
Which PDF version is this doc using and which PDFWriter was used?
This may be difficult but do you think you can provide me a sample file for further analysis?

**Comment by sannihith1995** [comment]
Hi @hhrutter,

Tried by returning '**0**' as well and it gives the same output.
`length: "Length" 256 not supported`

I am attaching the file with password - Chennai@2017
It has a csv attachment in it.
[[ENCRYPTED] attachments.pdf](https://github.com/hhrutter/pdfcpu/files/3235777/ENCRYPTED.attachments.pdf)

**Comment by hhrutter** [comment]
This PDF was produced by iText 2.1.7. It claims to be V1.4 but defacto using V1.7 Extension Level 3.
This extension level introduced AES 256 bit encryption.

Right now pdfcpu is supporting AES up to a key length of 128 bit this is inline with V1.7.
Stay tuned for an improved version supporting AES 256 bit.
Thanks for using pdfcpu 💚

**Comment by sannihith1995** [comment]
Thank you.. 👍😃

**Comment by hhrutter** [comment]
This is fixed with release [v0.1.24](https://github.com/hhrutter/pdfcpu/releases/tag/v0.1.24).
